jcl/src/java.base/share/classes/java/lang/invoke/GuardWithTestHandle.java

```diff
@@ -42,7 +42,15 @@ protected GuardWithTestHandle(MethodHandle guard, MethodHandle trueTarget, Metho
 		this.falseTarget = originalHandle.falseTarget;
 	}
 
-	public static GuardWithTestHandle get(MethodHandle guard, MethodHandle trueTarget, MethodHandle falseTarget) {
+	public static MethodHandle get(MethodHandle guard, MethodHandle trueTarget, MethodHandle falseTarget) {
+		if (guard instanceof ConstantIntHandle) {
```

Comment:
Can you add a comment explaining why this is valid?  Likely due to allowing ConstantHandles to drop unwanted arguments and implemementing booleans in terms of int.

Does this happen frequently though?  It seems like an unusual case